components/org.wso2.identity.event.http.publisher/src/main/java/org/wso2/identity/event/http/publisher/internal/component/HTTPAdapterServiceComponent.java

```diff
@@ -0,0 +1,130 @@
+import org.wso2.carbon.identity.core.util.IdentityCoreInitializedEvent;
+import org.wso2.carbon.identity.event.publisher.api.service.EventPublisher;
+import org.wso2.carbon.identity.webhook.management.api.service.WebhookManagementService;
+import org.wso2.carbon.identity.webhook.metadata.api.service.EventAdapterMetadataService;
+import org.wso2.identity.event.http.publisher.internal.config.HTTPAdapterConfiguration;
+import org.wso2.identity.event.http.publisher.internal.service.impl.HTTPEventPublisherImpl;
+
+import static org.wso2.identity.event.http.publisher.constant.HTTPAdapterConstants.HTTP_ADAPTER_NAME;
+
+/**
+ * HTTP Outbound Event Adapter service component.
+ */
+@Component(
+        name = "org.wso2.identity.event.http.publisher.internal.component.HTTPAdapterServiceComponent",
+        immediate = true)
+public class HTTPAdapterServiceComponent {
+
+    private static final Log log = LogFactory.getLog(HTTPAdapterServiceComponent.class);
+
+    @Activate
+    protected void activate(ComponentContext context) {
+
+        try {
+            HTTPAdapterDataHolder.getInstance().setAdapterConfiguration(new HTTPAdapterConfiguration(
+                    HTTPAdapterDataHolder.getInstance().getEventAdapterMetadataService()
+                            .getAdapterByName(HTTP_ADAPTER_NAME).getProperties()));
+            if (HTTPAdapterDataHolder.getInstance().getAdapterConfiguration().isAdapterEnabled()) {
+                // Register EventPublisher service
+                HTTPEventPublisherImpl eventPublisherService = new HTTPEventPublisherImpl();
+                context.getBundleContext().registerService(EventPublisher.class.getName(),
+                        eventPublisherService, null);
+                HTTPAdapterDataHolder.getInstance().setClientManager(new ClientManager());
+                log.debug("Successfully activated the HTTP adapter service.");
+            } else {
+                log.error("HTTP Adapter is not enabled.");
+            }
+        } catch (Throwable e) {
+            log.error("Can not activate the HTTP adapter service: " + e.getMessage(), e);
+        }
+    }
+
+    @Reference(
+            name = "identity.core.init.event.service",
+            service = IdentityCoreInitializedEvent.class,
+            cardinality = ReferenceCardinality.MANDATORY,
+            policy = ReferencePolicy.DYNAMIC,
+            unbind = "unsetIdentityCoreInitializedEventService"
+    )
+    protected void setIdentityCoreInitializedEventService(IdentityCoreInitializedEvent identityCoreInitializedEvent) {
+        /* reference IdentityCoreInitializedEvent service to guarantee that this component will wait until identity core
+         is started */
+    }
+
+    protected void unsetIdentityCoreInitializedEventService(IdentityCoreInitializedEvent identityCoreInitializedEvent) {
+        /* reference IdentityCoreInitializedEvent service to guarantee that this component will wait until identity core
+         is started */
+    }
```

Comment:
Is this necessary ?

Reply:
fixed with https://github.com/wso2-extensions/identity-event-publishers/pull/37/commits/15538dc2b9ce588c47e8db9c5db39156de457b3c